src/main/scala/io/iohk/ethereum/blockchain/sync/fast/PivotBlockSelector.scala

```diff
@@ -188,12 +210,20 @@ class PivotBlockSelector(
     val bestPeerBestBlockNumber = peersSortedByBestNumber.headOption
       .map { case (_, bestPeerBestBlockNumber) => bestPeerBestBlockNumber }
       .getOrElse(BigInt(0))
-    val expectedPivotBlock = (bestPeerBestBlockNumber - syncConfig.pivotBlockOffset).max(0)
+
+    // The current best block number is pushed back by `pivotBlockNumberResetDelta`
+    // if this request is issued by the retry logic.
+    val currentBestBlockNumber: BigInt =
+      previousBestBlockNumber
+        .map(_ - pivotBlockNumberResetDelta.max(0))
+        .getOrElse(bestPeerBestBlockNumber)
+
+    val expectedPivotBlock = (currentBestBlockNumber - syncConfig.pivotBlockOffset).max(0)
     val correctPeers = peersSortedByBestNumber
       .takeWhile { case (_, number) => number >= expectedPivotBlock }
       .map { case (peer, _) => peer }
 
-    ElectionDetails(correctPeers, expectedPivotBlock)
+    ElectionDetails(correctPeers, currentBestBlockNumber, expectedPivotBlock)
```

Comment:
I think there's a nice opportunity to split some of the functionality in this file out into a couple pure functions / methods involving the `ElectionDetails` type. We could add some unit tests for it that way.